internal/backend/runtime/omni/controllers/omni/internal/exposedservice/reconciler.go

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) 2025 Sidero Labs, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+
+package exposedservice
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/base64"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/siderolabs/gen/optional"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/siderolabs/omni/client/pkg/omni/resources"
+	"github.com/siderolabs/omni/client/pkg/omni/resources/omni"
+)
+
+// Reconciler is the reconciler for ExposedService resources.
+type Reconciler struct {
+	r      controller.ReaderWriter
+	logger *zap.Logger
+
+	exposedServices map[resource.ID]*omni.ExposedService
+	usedAliases     map[string]resource.ID
+
+	cluster                string
+	workloadProxySubdomain string
+	advertisedAPIURL       string
+
+	services []*corev1.Service
+}
```

Comment:
Here is no logical change - I simply gather all required variables into `Reconciler` to avoid passing them as function args.